### PR TITLE
Django

### PR DESCRIPTION
**Pull request by zhongchen530**
I have restructured my project as a django project and have written script to write all current elos and historical elos of each team on a local database. The next step might be to replace local database with RDS and write elos for each player.

### CODE REVIEW

**Review by darekj28 — CHANGES_REQUESTED** [review]
Please add .ini, .pyc  and .DS_store files to your .gitignore so they don't show up in PR.s

File: NBA/admin.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from .models import Game,Current_team_elo,Historical_team_elo
+admin.site.register(Game)
```
Comment (darekj28) [inline NBA/admin.py:3]: What does this do? Can you add comments explaining?

File: NBA/getGames.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from nba_api.stats.static import teams,players
+from nba_api.stats.endpoints import leaguegamefinder
```
Comment (darekj28) [inline NBA/getGames.py:2]: Let's focus on your naming conventions.

For all file names use `snake_case` 

for class names use `PascalCase` 

for all variable names use `snake_case`

File: NBA/getGames.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import pandas as pd
+
+team_columns = ['TEAM_ID', 'GAME_ID',
```
Comment (darekj28) [inline NBA/getGames.py:5]: This is hard coded and we should avoid magic strings like this as much as possible.

File: NBA/wsgi.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+It exposes the WSGI callable as a module-level variable named ``application``.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/4.2/howto/deployment/wsgi/
+"""
+
+import os
+
+from django.core.wsgi import get_wsgi_application
+
```
Comment (darekj28) [inline NBA/wsgi.py:13]: What's different about this and asgi .py?

File: manage.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+"""Django's command-line utility for administrative tasks."""
+import os
+import sys
+
+
+def main():
```
Comment (darekj28) [inline manage.py:7]: What is the purpose of this file? If it's django boilerplate, please move all of the boilerplate functions to a folder called `django` .

File: NBA/settings.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+Generated by 'django-admin startproject' using Django 4.2.1.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/4.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/4.2/ref/settings/
+"""
+
```
Comment (darekj28) [inline NBA/settings.py:12]: If this is auto generated, can you add it to a separate django folder for all the django setup

File: NBA/models.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    game_id = models.IntegerField(primary_key = True)
+    team_id = models.IntegerField()
+    game_date = models.CharField(max_length=15)
+    wl = models.CharField(max_length=1)
+    opponent_id = models.IntegerField()
+
+    def __str__(self):
+        return str(self.game_id) + " " + self.game_date
+
+class Historical_team_elo(models.Model):
```
Comment (darekj28) [inline NBA/models.py:13]: Change name to pascalcase.

All of these models should be moved to a `models` folder in separate files

game.py
team_elo.py


Have you considered combining historical and current team elo into a single object and use an instance variable to distinguish them.

File: scripts/InitialCompute.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
+from NBA import getGames
```
Comment (darekj28) [inline scripts/InitialCompute.py:1]: rename filename to snake_case

File: scripts/InitialCompute.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+def write_games():
+    date_range = ["2000-00-00","9999-99-99"]
+    teams_df = getGames.get_team_games(date_range)
+    print(teams_df)
+    for indx,row in teams_df.iterrows():
+        object = Game(*(row.tolist()))
+        object.save()
+
+def elo_diff(elo1,elo2,wl):
```
Comment (darekj28) [inline scripts/InitialCompute.py:13]: add comments and typing explaining how this elo works.

File: scripts/InitialCompute.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    p1_win = 1/(1+10**((elo2-elo1)/400))
+    if wl == "W":
```
Comment (darekj28) [inline scripts/InitialCompute.py:15]: If you have your own Game object, this function should take in 2 Teams (which includes their elo) and a Game then update their Elo and the Players Elo accordingly after.

File: scripts/InitialCompute.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+        score = 1
+    elif wl == "L":
+        score = 0
+    elif wl == "T":
+        score = 0.5
+
+    return 32*(score - p1_win)
+
+def write_elo():
```
Comment (darekj28) [inline scripts/InitialCompute.py:24]: Can you explain what this function does?

It seems like this just writes to the database right?

File: scripts/InitialCompute.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    games = Game.objects.order_by("-game_date")
```
Comment (darekj28) [inline scripts/InitialCompute.py:25]: Try to avoid magic strings like this whenever possible.

File: NBA/getGames.py [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+       'GAME_DATE', 'MATCHUP', 'WL']
+
```
Comment (darekj28) [inline NBA/getGames.py:7]: This file does everything and it should be broken up.

You should have one file called `nba_api_helper.py` which contains a class `NBA_ApiHelper` which has functions like `get_games` and gets the raw list of games form NBA API based on query parameters. 

Then you should have a file that has functions to clean and reformat the data in preparation for computing elo. This file needs tests. 

Lastly, you can have a separate file with functions to write your resultant data to the database when you update the elos. This takes in the raw data and returns the data in a friendly format for computing ELOs.

This allows you to test the code that does the data clean without being dependent on APIs.

**Review by darekj28 — CHANGES_REQUESTED** [review]

File: NBA/getGames.py [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+        for team in teams.get_teams():
+            name_to_id[team["full_name"]] = team["id"]
+    return name_to_id[team_name]
+
+id_to_name = {}
+def get_team_name(team_id):
+    if not id_to_name:
+        for team in teams.get_teams():
+            id_to_name[team["id"]] = team["full_name"]
+    print(id_to_name,"heyheyheyheyhey")
```
Comment (darekj28) [inline NBA/getGames.py:20]: remove the random print statements

**Review by darekj28 — CHANGES_REQUESTED** [review]

File: scripts/InitialCompute.py (hunk 11 above)
Comment (zhongchen530) [reply scripts/InitialCompute.py:24]: This function compute the historical and current elos for each team and wrote them to database using the games table we obtained from the api earlier.

File: manage.py (hunk 5 above)
Comment (zhongchen530) [reply manage.py:7]: I think for all django project, the manage.py file is where it needs to be when it is generated?

File: NBA/settings.py (hunk 6 above)
Comment (zhongchen530) [reply NBA/settings.py:12]: This is auto generated by django. Not sure if moving it will mess up the setup.

**Pull request closed without merging**